packages/gatsby-theme-styleguide/gatsby-node.js

```diff
@@ -22,45 +22,48 @@ exports.onCreateNode = ({ node, getNode, actions }) => {
   }
 }
 
-exports.createPages = ({ graphql, actions }) => {
+exports.createPages = async ({ graphql, actions, reporter }) => {
+  // Destructure the createPage function from the actions object
   const { createPage } = actions
-
-  return new Promise((resolve, reject) =>
-    graphql(`
-      {
-        allMarkdownRemark {
-          edges {
-            node {
-              fields {
-                slug
-              }
+  const result = await graphql(`
+    query {
+      allMdx {
+        edges {
+          node {
+            id
+            fields {
+              slug
             }
           }
         }
       }
-    `)
-      .then(res => {
-        res.data.allMarkdownRemark.edges.map(({ node }) => {
-          createPage({
-            path: node.fields.slug,
-            component: path.join(
-              __dirname,
-              'src',
-              'templates',
-              'pattern-page.js'
-            ),
-            context: {
-              // Data passed to context is available
-              // in page queries as GraphQL variables.
-              slug: node.fields.slug,
-            },
-          })
-        })
+    }
+  `)
+  if (result.errors) {
+    reporter.panicOnBuild('🚨  ERROR: Loading "createPages" query')
+  }
+
+  const posts = result.data.allMdx.edges
+
+  posts.forEach(({ node }) => {
+    createPage({
+      path: node.fields.slug,
+      component: path.join(
+        __dirname,
+        'src',
+        'templates',
+        'pattern-page-mdx.js'
+      ),
+      context: {
+        id: node.id,
+        slug: node.fields.slug,
+      },
+    })
+  })
 
-        resolve()
-      })
-      .catch(err => {
-        reject(err)
-      })
-  )
+  // creating root index page
+  createPage({
+    path: '/',
+    component: path.join(__dirname, '../../', 'src', 'docs', 'index.js'),
```

Comment:
keeping the pages directory is not working ? 
If it does, I would prefer this option because we may have other pages later that are not fully docs, and keeping them outside of the mdx files could be good

Reply:
Yes, It works on src/pages, I'm gonna keep the index.[js,scss] there.